### PR TITLE
No double-to-float conversions

### PR DESCRIPTION
**Pull request by serin-delaunay**
Corner weights from `TilePartitionerGradient` and `TilePartitionerPerlin` were being converted from doubles to floats before `AlphaCalculator` methods were applied. I don't think there's any good reason to do that. This PR removes all of those conversions.

These conversions were a suspected cause of #51. The issue remains unresolved, but the size and frequency of black patches may have been reduced.

### CODE REVIEW

File: Wangscape/tilegen/partition/TilePartitionerGradient.cpp [hunk 1]
```diff
@@ -25,18 +25,18 @@ void TilePartitionerGradient::makePartition(TilePartition & regions,
...
         for (size_t y = 0; y < mOptions.tileFormat.resolution; y++)
         {
-            weights[0] = (float)gradientWeight(x, y, 0, 0, quarter_res);
-            weights[1] = (float)gradientWeight(x, y, 0, resolution_sub_1, quarter_res);
-            weights[2] = (float)gradientWeight(x, y, resolution_sub_1, 0, quarter_res);
-            weights[3] = (float)gradientWeight(x, y, resolution_sub_1, resolution_sub_1, quarter_res);
+            weights[0] = (double)gradientWeight(x, y, 0, 0, quarter_res);
+            weights[1] = (double)gradientWeight(x, y, 0, resolution_sub_1, quarter_res);
+            weights[2] = (double)gradientWeight(x, y, resolution_sub_1, 0, quarter_res);
+            weights[3] = (double)gradientWeight(x, y, resolution_sub_1, resolution_sub_1, quarter_res);
```
Comment (hryniuk) [inline Wangscape/tilegen/partition/TilePartitionerGradient.cpp]: I'm not sure if it must be casted, but if it does, use `static_cast<double>` here, please. One advantage is that it's easy to grep for, and the other is that it is safer (e.g. another version of "raw casting", `double(...)` can be treated by a compiler as a function declaration).

Comment (serin-delaunay) [inline Wangscape/tilegen/partition/TilePartitionerGradient.cpp]: Done.

File: Wangscape/tilegen/partition/TilePartitionerPerlin.cpp [hunk 2]
```diff
@@ -29,16 +29,16 @@ Reseedable TilePartitionerPerlin::makeCornerModule(const Corners& corners,
...
 
-void TilePartitionerPerlin::noiseToAlpha(std::vector<noise::RasterValues<float>>& noise_values,
+void TilePartitionerPerlin::noiseToAlpha(std::vector<noise::RasterValues<double>>& noise_values,
                                          std::vector<sf::Image>& outputs,
                                          size_t resolution) const
 {
-    std::vector<float> weights((int)CORNERS);
-    alpha::AlphaCalculatorMax ac;
+    std::vector<double> weights((int)CORNERS);
+    alpha::AlphaCalculatorLinear ac;
```
Comment (hryniuk) [inline Wangscape/tilegen/partition/TilePartitionerPerlin.cpp]: Why did you change `AlphaCalculatorMax` to `AlphaCalculatorLinear`?

Comment (serin-delaunay) [inline Wangscape/tilegen/partition/TilePartitionerPerlin.cpp]: #51 Only manifests when `AlphaCalculatorLinear` is used. I changed to `AlphaCalculatorLinear` in order to observe the effects of the changes in that use case.

In future the choice of `AlphaCalculator` should be in the options file (#37); at the current stage of the project I don't think it's very important which one is hard-coded.
